common/flatpak-dir.c

```diff
@@ -12586,3 +12681,69 @@ flatpak_dir_get_source_pid (FlatpakDir *self)
   return self->source_pid;
 }
 
+static void
+(flatpak_dir_log) (FlatpakDir *self,
+                   const char *file,
+                   int line,
+                   const char *func,
+                   const char *source, /* overrides self->name */
+                   const char *change,
+                   const char *remote,
+                   const char *ref,
+                   const char *commit,
+                   const char *url,
+                   const char *format,
+                 ...)
+{
+#ifdef HAVE_LIBSYSTEMD
+  const char *installation = source ? source : flatpak_dir_get_name_cached (self);
+  pid_t source_pid = flatpak_dir_get_source_pid (self);
+  char message[1024];
+  int len;
+  va_list args;
+
+  len = g_snprintf (message, sizeof (message), "%s: ", installation);
+
+  va_start (args, format);
+  g_vsnprintf (message + len, sizeof (message) - len, format, args);
+  va_end (args);
+
+  /* See systemd.journal-fields(7) for the meaning of the
+   * standard fields we use, in particular OBJECT_PID
+   */
+  if (source_pid)
```

Comment:
The duplication here is because I'm not 100% sure if setting OBJECT_PID to 0 is going to confuse journald.